### ISSUE TITLE
"Always open in this container" is completely broken

### ISSUE THREAD

**Issue by PureTryOut** [mozilla/testpilot-containers#678]
Besides last update making it (in my opinion) way less obvious how to open a tab in a certain container and that menu itself looking way uglier than it used too be, the "always open in this container" feature is now completely broken for me.

When I visit a site which is set to open in a specific container, Firefox asks me to open it in the current container or the container I specified. When pressing either of the buttons, that confirmation page just reloads and now asks me the same thing for the page "null".

![screenshot_2017-07-16_11-50-55](https://user-images.githubusercontent.com/1565836/28246557-0a800942-6a1d-11e7-8f4e-c0d1e77d5b13.png)

So whenever I open a link from anywhere, and that page opens up, I have to manually open a new tab in the container I need it in, and then paste the link in. This makes the addon **really** annoying to use, and I rather disable it to be honest. Does anyone else have this issue? It happens on both my laptop and desktop running Firefox 52.2.0 on Linux.

**Comment by jonathanKingston** [comment]
Hey @PureTryOut I was able to see this in the development stage of this feature however the bug was fixed and relating to a storage issue of the assignment.

I'm not exactly sure why 52 would do this however we no longer support ESR firefox unfortunately as we rely on new APIs that are only available in 53: https://blog.mozilla.org/addons/2017/01/25/webextensions-in-firefox-53/

As mentioned here: https://github.com/mozilla/testpilot-containers/issues/674#issuecomment-315451280

> yeah we are migrating more and more to WebExtensions, some of which are not available in the ESR unfortunately. Backporting patches to ESR isn't a huge priority due to the numbers however it might be something we could do later because of the Tor population.
>
> Anyway... I would advise ESR users uninstalling the extension through test pilot and using nightly. In about:config change the following after uninstall:
>
> - privacy.userContext.enabled = true
> - privacy.userContext.ui.enabled = true
> - privacy.userContext.longPressBehavior = 2
>
> It is likely this is how they currently are with the extension set up. To modify containers you can go to about:preferences#containers which is navigatable via the Privacy panel.
Basically this is a less feature rich version of what we have here but will keep working for all of the ESR period.
>
> For the time being I am going to close this issue as we have already agreed on dropping support. I'm sorry about this I know it's not the best position.

We may consider shipping a separate addon that  52 users could use due to the time scale to get to 59. However currently we don't have a plan to do this.

**Comment by PureTryOut** [comment]
That did indeed fix it. I was planning to stay on the most recent stable build of Firefox what my distribution offered, I guess not. Thanks for your help!

**Comment by jonathanKingston** [comment]
@kjozwiak this was the bug I mentioned in the meeting being reproducible sometimes even in nightly despite this being about ESR.
It could be funky URLs but I don't thik it is, it's more likely corrupt storage on update from an older version perhaps.
I saw this on a work wiki tab linking to a bugzilla link which is assigned to bugzilla. The link itself I don't think even had params other than the bug ID. When I clicked open in bugzilla it went to null. However after restarting I wasn't able to repro with the same link even.
